### DISCUSSION TITLE
How to use URL params in tRPC? Like - /api/users/:userId/account

### DISCUSSION THREAD

**Discussion by hmShuvo314** [trpc/trpc#3763] [General]
Hello, I have been using REST for years but am new to using tRPC. I just have couple of questions.

Say I have an API route similar to - **/api/users/:userId/account**

In REST API using express, for example, we could use something like **req.params.name** to get the **userId**, now I wonder how do I do similar things with tRPC. 

My experience with tRPC has been awesome. But I just want to know if I can use same REST standards for using URL params with tRPC and whether it will be any problem if I surface my tRPC APIs to the public and they want to access it without the tRPC client (say they want to use Axios, for example)?

Thanks in advance!

**Comment by KATT** [comment]
If you want something REST-esque, you can use https://github.com/jlalmes/trpc-openapi

**Reply by hmShuvo314** [reply]
Thanks for replying! But then how do I get the end-to-type checking?

**Reply by KATT** [reply]
Something like orval
